roles/metallb/tasks/main.yml

```diff
@@ -0,0 +1,48 @@
+---
+# NOTE(gyee): we'll only include the first one that matches. The rest will be ignored.
+- name: Include OS family vars
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_os_family }}.yml"
+    - "{{ ansible_os_family | lower }}.yml"
+    - "main.yml"
+
+- name: Include OS distribution vars
+  include_vars: "{{ item }}"
+  with_first_found:
+    - "{{ ansible_distribution | lower | replace(' ', '_') }}.yml"
+    - "{{ ansible_distribution | replace(' ', '_') }}.yml"
+    - "main.yml"
+
+- name: Include platform-specific common tasks
+  include_tasks: "_{{ ansible_os_family | lower }}.yml"
+
+- name: Add metallb helm repo
+  kubernetes.core.helm_repository:
+    repo_url: "{{ metallb_helm_repo_url }}"
+    name: "{{ metallb_helm_repo_name }}"
+
+- name: Deploy metallb helm chart
+  shell: |
+    env HELM_EXPERIMENTAL_OCI=1 \
+    helm upgrade --install {{ metallb_release }} {{ metallb_helm_chart_ref }} \
+      --create-namespace --namespace {{ metallb_namespace }} \
+      --atomic --timeout 10m
```

Comment:
I know this is the same as other roles, but is there a reason we need to use `shell` instead of `kubernetes.core.helm` here?

Reply:
I think no, kubernetes.core.helm should work as well here but just wanted to be consistent with the other roles